zathura/utils.c

```diff
@@ -257,3 +257,26 @@ GdkAtom* get_selection(zathura_t* zathura)
 
   return selection;
 }
+
+unsigned int find_first_page_column(const char* first_page_column_list,
+                                    const unsigned int pages_per_row) {
+  /* copy settings list */
+  char* settings = malloc(strlen(first_page_column_list));
+  strcpy(settings, first_page_column_list);
```

Comment:
The buffer is not large enough. Please just use `g_strdup` here. Also, the buffer is never freed.